src/context-providers/builder/provider.tsx

```diff
@@ -17,6 +17,7 @@ const DEFAULT_VALUES: IBuilderState = {
     focusedElement: null,
     showSidePanel: true,
     orderedIdentifiers: [],
+    pastMode: {},
```

Comment:
as the provider can store various states not limited to the side panel, we will need to be more specific with the naming, suggest to use `pastSidePanelState`

Reply:
I do not think it is a good idea to keep reference to the past state. Maybe we should decouple showing the edit panel as a `mode`. Instead we can use the `focusedElement` as the reference. If it is not null, it means we will show the edit element panel.

Likewise for the side panel, if `focusedElement`is not null, show the side panel. Then the original `showSidePanel` should still be holding the value prior to focusing the element.